File: Week -3/Qs9.py
import sys

def copy_file(source_filename, destination_filename):
    try:
        with open(source_filename, 'rb') as source_file:
            with open(destination_filename, 'wb') as destination_file:
                while True:
                    buffer = source_file.read(1024)
                    if not buffer:
                        break
                    destination_file.write(buffer)
        print(f"File copied successfully from {source_filename} to {destination_filename}")
    except FileNotFoundError as e:
        print(f"Error: {e}")
    except IOError as e:
        print(f"Error: {e}")

def main():
    if len(sys.argv) != 3:
        print("Usage: python copy_file.py <source_file> <destination_file>")
        sys.exit(1)

    source_filename = sys.argv[1]
    destination_filename = sys.argv[2]

    copy_file(source_filename, destination_filename)

if __name__ == "__main__":
    main()

File: Week -3/Qs9.c
#include <stdio.h>
#include <stdlib.h>

#define BUFFER_SIZE 1024

int main(int argc, char *argv[]) {
    if (argc != 3) {
        fprintf(stderr, "Usage: %s <source_file> <destination_file>\n", argv[0]);
        return 1;
    }

    const char *sourceFilename = argv[1];
    const char *destinationFilename = argv[2];

    FILE *sourceFile = fopen(sourceFilename, "rb");
    if (sourceFile == NULL) {
        perror("Error opening source file");
        return 1;
    }

    FILE *destinationFile = fopen(destinationFilename, "wb");
    if (destinationFile == NULL) {
        perror("Error opening destination file");
        fclose(sourceFile);
        return 1;
    }

    char buffer[BUFFER_SIZE];
    size_t bytesRead;

    // Read from source and write to destination
    while ((bytesRead = fread(buffer, 1, BUFFER_SIZE, sourceFile)) > 0) {
        fwrite(buffer, 1, bytesRead, destinationFile);
    }

    // Close both files
    fclose(sourceFile);
    fclose(destinationFile);

    printf("File copied successfully from %s to %s\n", sourceFilename, destinationFilename);

    return 0;
}
